core/src/main/java/com/linecorp/armeria/server/AbstractRouteBuilder.java

```diff
@@ -0,0 +1,526 @@
+import static com.linecorp.armeria.common.HttpMethod.DELETE;
+import static com.linecorp.armeria.common.HttpMethod.GET;
+import static com.linecorp.armeria.common.HttpMethod.HEAD;
+import static com.linecorp.armeria.common.HttpMethod.OPTIONS;
+import static com.linecorp.armeria.common.HttpMethod.PATCH;
+import static com.linecorp.armeria.common.HttpMethod.POST;
+import static com.linecorp.armeria.common.HttpMethod.PUT;
+import static com.linecorp.armeria.common.HttpMethod.TRACE;
+import static com.linecorp.armeria.server.ServerConfig.validateMaxRequestLength;
+import static com.linecorp.armeria.server.ServerConfig.validateRequestTimeoutMillis;
+import static java.util.Objects.requireNonNull;
+
+import java.nio.charset.Charset;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.function.Function;
+
+import javax.annotation.Nullable;
+
+import com.google.common.collect.ImmutableList;
+
+import com.linecorp.armeria.common.HttpMethod;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.common.logging.ContentPreviewer;
+import com.linecorp.armeria.common.logging.ContentPreviewerFactory;
+import com.linecorp.armeria.internal.ArmeriaHttpUtil;
+
+/**
+ * A builder class for binding a {@link Service} fluently.
+ *
+ * @param <B> the self type
+ * @param <V> {@link ServerBuilder} or {@link VirtualHostBuilder} that you create this builder from
+ *
+ * @see RouteBuilder
+ * @see VirtualHostRouteBuilder
+ */
+public abstract class AbstractRouteBuilder<B extends AbstractRouteBuilder<B, V>, V> {
+
+    private static final EnumSet<HttpMethod> defaultMethods =
+            EnumSet.of(OPTIONS, GET, HEAD, POST, PUT, PATCH, DELETE, TRACE, CONNECT);
+
+    @Nullable
+    private PathMapping defaultPathMapping;
+
+    private Set<HttpMethod> methods = defaultMethods;
+
+    private final Map<PathMapping, Set<HttpMethod>> pathMappings = new HashMap<>();
+
+    private List<MediaType> consumeTypes = ImmutableList.of();
+    private List<MediaType> produceTypes = ImmutableList.of();
+
+    @Nullable
+    private Long requestTimeoutMillis;
+    @Nullable
+    private Long maxRequestLength;
+    @Nullable
+    private Boolean verboseResponses;
+    @Nullable
+    private ContentPreviewerFactory requestContentPreviewerFactory;
+    @Nullable
+    private ContentPreviewerFactory responseContentPreviewerFactory;
+    @Nullable
+    private Function<Service<HttpRequest, HttpResponse>, Service<HttpRequest, HttpResponse>> decorator;
+
+    @SuppressWarnings("unchecked")
+    final B self() {
+        return (B) this;
+    }
+
+    /**
+     * Sets the path pattern that the {@linkplain #service(Service) service} is bound to. The {@link Service}
```

Comment:
- the service -> `a {@link Service}` ?
- is bound to -> will be bound to ?

Reply:
Fixed, thanks!